src/system/version_migration/version_converter.ts

```diff
@@ -21,12 +33,12 @@ const LATEST_VERSION = version.split(".").map((value) => parseInt(value));
  * @see {@link SystemVersionConverter}
  */
 export function applySystemVersionMigration(data: SystemSaveData) {
-  const curVersion = data.gameVersion.split(".").map((value) => parseInt(value));
+  const prevVersion = data.gameVersion;
 
-  if (!curVersion.every((value, index) => value === LATEST_VERSION[index])) {
+  if (compareVersions(prevVersion, version) === -1) {
```

Comment:
```suggestion
  if (compareVersions(prevVersion, version) < 0) {
```
I think I prefer this?

Reply:
The compare function can only return `1 | 0 | -1`, where `1` is "the first version is higher", `0` is "the versions are equal", and `-1` is "the second version is higher". It works the same as various built-in JS/TS functions for comparison.

Reply:
can you assign this to a variable with an appropriate name for context and use that variable in the condition?

```suggestion
  var isCurrentVersionHigher = compareVersions(prevVersion, version) === -1;
  if (isCurrentVersionHigher) {
```

Reply:
For me `<0` / `>0` is easier to understand than `=== (-)1`, but using a const is actually much clearer

Reply:
I looked at the other functions the module provides and there's also `compare(versionA, versionB, comparisonOperator)` which lets you do a specific comparison (ie: `compare(versionA, versionB, "<")` would return `true` if `versionA` is less than `versionB`). Would that be better or should I go with the const?

Reply:
That' seems fine yeah, I don't have any strong preference for one or the other